lib/cli/schemas/replication-state-schema.js

```diff
@@ -0,0 +1,17 @@
+const Joi = require('@hapi/joi')
+
+const schema = Joi.object({
+  replicaSet: Joi.object({
+    replicaSetName: Joi.string().required(),
+    servers:
+     Joi.array().min(1).required()
+       .items(Joi.object({
+         host: Joi.string().required(),
+         uri: Joi.string().uri()
```

Comment:
https://hapi.dev/module/joi/api/?v=17.1.1#stringurioptions we can also validate mongo:// is the only schemee

Reply:
btw joi can also default a value to a child value (this way we can remove  those hacks from our libs) 

But I  can't  find the link now

ie you can do `uri.defaults((obj) -> `mongodb://${object.host}`)` something like  this but  the syntax might be off